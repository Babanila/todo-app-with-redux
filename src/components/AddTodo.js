import React, { useState } from 'react'
import { useDispatch, connect } from 'react-redux'
import { cx, css } from 'emotion'
import { addTodo } from '../actions'

const addTodoDivStyles = css`
  width: 100%;
  display: flex;
  justify-content: center;
`

const formStyles = css`
  width: 80%;
  display: flex;
  justify-content: center;
`

const inputStyles = css`
  width: 60%;
  height: 2em;
  border: 1px solid #000000;
  border-radius: 5px;
  padding-left: 1em;
`
const addButtonStyles = css`
  font-size: 16px;
  color: #ffffff;
  background: #0000ff;
  border: 1px solid #0000ff;
  border-radius: 5px;
  padding: 0 10px;
  margin-left: 5px;
  outline: 0;
  &:hover {
    cursor: pointer;
    background: #ffffff;
    color: #000000;
    border: 1px solid #000000;
  }
`

function AddTodo() {
  const [input, setInput] = useState('')
  const handleInputChange = (item) => setInput(item)
  const dispatch = useDispatch()

  const handleCreateTodo = (e) => {
    e.preventDefault()
    if (input) dispatch(addTodo(input))
    setInput('')
  }
  return (
    <div className={cx(addTodoDivStyles)}>
      <form className={cx(formStyles)} onSubmit={handleCreateTodo}>
        <input
          placeholder='What needs to be done?...'
          className={cx(inputStyles)}
          value={input}
          onChange={(e) => handleInputChange(e.target.value)}
        />
        <button className={cx(addButtonStyles)} type='submit' onClick={handleCreateTodo}>
          Add
        </button>
      </form>
    </div>
  )
}
export default connect(null, { addTodo })(AddTodo)
