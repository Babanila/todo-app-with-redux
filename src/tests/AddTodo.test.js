import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import AddTodo from '../components/AddTodo'

const mockStore = configureStore([thunk])

describe('AddTodo', () => {
  let store, AddTodoWrapper

  beforeEach(() => {
    store = mockStore({})

    AddTodoWrapper = mount(
      <Provider store={store}>
        <AddTodo />
      </Provider>
    ).mount()
  })

  it('should be defined', () => {
    expect(AddTodo).toBeDefined()
  })

  it('should be a function', () => {
    expect(typeof AddTodo).toBe('function')
  })

  it('should renders site title correctly', () => {
    expect(AddTodoWrapper).toMatchSnapshot()
  })

  it('should render the exact number of element in component', () => {
    expect(AddTodoWrapper.find('div').length).toEqual(1)
    expect(AddTodoWrapper.find('form').length).toEqual(1)
    expect(AddTodoWrapper.find('input').length).toEqual(1)
    expect(AddTodoWrapper.find('button').length).toEqual(1)
  })

  it('<input> updates input value on onChange event', () => {
    const eventObj = { target: { value: 'Review Code' } }
    expect(AddTodoWrapper.find('input').prop('value')).toEqual('')

    AddTodoWrapper.find('input').simulate('change', eventObj)
    AddTodoWrapper.update()
    expect(AddTodoWrapper.find('input').prop('value')).toEqual('Review Code')
  })
})
