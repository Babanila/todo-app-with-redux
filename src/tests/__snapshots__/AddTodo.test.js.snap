// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should renders site title correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AddTodo>
    <div
      className="css-1hgts9t"
    >
      <form
        className="css-roxwtq"
        onSubmit={[Function]}
      >
        <input
          className="css-xdfs90"
          onChange={[Function]}
          placeholder="What needs to be done?..."
          value=""
        />
        <button
          className="css-1rs6h6a"
          onClick={[Function]}
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </AddTodo>
</Provider>
`;
